### DISCUSSION TITLE
space() problem?

### DISCUSSION THREAD

**Discussion by BattlemasterLoL** [zauberzeug/nicegui#2579] [Q&A]
I am working on a footer for my ui and ran into a problem. It is looking like the `space()` [function](https://nicegui.io/documentation/space) is not working unless I am missing something. Any tips on getting around this or using another method? Here is what I am working with so far:
```py
    with ui.footer().style("background-color: #26272b;").classes('w-full'):
        with ui.row().style("align-self: left; align-items: center; height: 50px"):
            ui.html(
                'Copyright <a href="#" target="_blank" style="color: #6FA5D8;">Company</a>.'
            )
            ui.space()
            with ui.grid(columns=4).style('align-self: right;'):
                ui.label("t1")
                ui.label("t2")
                ui.label("t3")
                ui.label("t4")
```

**Comment by rodja** [comment]
Simply add `.classes('w-full')` to the `ui.row()`. Otherwise it does not expand to the full width.

**Reply by BattlemasterLoL** [reply]
Does not work for me. The html and grid remain to the left and side by side. I need space between them where the html is to the left and grid to the right.

**Reply by rodja** [reply]
If I use this code:

```py
@ui.page('/')
async def index():
    with ui.footer().style("background-color: #26272b;").classes('w-full'):
        with ui.row().style("align-self: left; align-items: center; height: 50px").classes('w-full'):
            ui.html(
                'Copyright <a href="#" target="_blank" style="color: #6FA5D8;">Company</a>.'
            )
            ui.space()
            with ui.grid(columns=4).style('align-self: right;'):
                ui.label("t1")
                ui.label("t2")
                ui.label("t3")
                ui.label("t4")
```

I'll get

<img width="598" alt="Screenshot 2024-02-21 at 03 41 19" src="https://github.com/zauberzeug/nicegui/assets/131391/af2b55d2-3f67-4fda-8b43-b7b26ecc7a0f">

Is the result different for you? Or have I misunderstood your requirement.

**Reply by BattlemasterLoL** [reply]
That is what I am trying to accomplish but I get the following error:
![image](https://github.com/zauberzeug/nicegui/assets/85181605/d001355a-3fa2-43c8-87af-1f9584999fac)

**Reply by BattlemasterLoL** [reply]
After doing some more testing this looks to be a problem only when running the code using VScode as my IDE. Running the script through command line works fine.

**Reply by rodja** [reply]
Ah, then verify that you have selected the right Python environment in the IDE. Maybe there is an old version of NiceGUI installed?
